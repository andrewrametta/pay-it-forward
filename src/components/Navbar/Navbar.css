nav.navbar-wrapper {
  height: 10vh;
  background: white;
  display: flex;
  align-items: baseline;
  position: relative;
}

.button {
  background: #4500c9;
  color: white;
}

.logo {
  flex: 1;
  margin: 15px;
}

.logo-image {
  max-width: 240px;
}

.nav-links {
  display: flex;
  width: 70%;
  height: 100%;
  justify-content: space-around;
  align-items: center;
  margin-left: auto;
}

.nav-links li a {
  color: #4500c9;
  text-decoration: none;
  font-size: 16px;
}

@media all and (max-width: 900px) {
  .line {
    width: 30px;
    height: 3px;
    background: #4500c9;
    margin: 5px;
  }

  nav {
    position: relative;
  }

  .logo-image {
    max-width: 200px;
  }

  .logo {
    margin: 15px;
    vertical-align: baseline;
  }

  .hamburger {
    position: absolute;
    cursor: pointer;
    right: 5%;
    top: 50%;
    transform: translate(-5%, -50%);
    z-index: 2;
  }

  .nav-links {
    position: fixed;
    background: white;
    height: 100vh;
    width: 100%;
    flex-direction: column;
    clip-path: circle(100px at 90% -10%);
    -webkit-clip-path: circle(100px at 90% -10%);
    transition: all 1s ease-out;
    pointer-events: none;
  }
  .nav-links.open {
    clip-path: circle(1500px at 90% -10%);
    -webkit-clip-path: circle(1500px at 90% -10%);
    pointer-events: all;
  }

  .nav-links li {
    opacity: 0;
    transition: all 0.5s ease 0.3s;
  }

  .nav-links li a {
    font-size: 25px;
  }

  li.fade {
    opacity: 1;
  }
}
