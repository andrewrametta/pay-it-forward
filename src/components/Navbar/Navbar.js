import React, { useContext } from "react";
import { Link } from "react-router-dom";
import TokenService from "../../services/token-service";
import AppContext from "../../AppContext";
import "./Navbar.css";

function Navbar(props) {
  const { type, setType, isLogged, setIsLogged } = useContext(AppContext);
  const logout = () => {
    TokenService.clearAuthToken();
    TokenService.clearUserType();
    setIsLogged(null);
    setType(null);
    props.history.push("/");
  };

  return (
    <div className="navbar-wrapper">
      <Link className="logo" to="/">
        <h1>PayItForward</h1>
      </Link>
      <Link to="/about">
        <button>About</button>
      </Link>
      {isLogged ? (
        <>
          <Link to="/dashboard">
            <button>Dashboard</button>
          </Link>
          {type == "user" ? (
            <Link to="/newdonation">
              <button>New Donation</button>
            </Link>
          ) : null}

          <Link to="/messages">
            <button>Messages</button>
          </Link>
          <button
            className="button"
            aria-label="logout-button"
            onClick={logout}
          >
            Logout
          </button>
        </>
      ) : (
        <>
          <Link to="/register">
            <button>Register</button>
          </Link>
          <Link to="/login">
            <button>Login</button>
          </Link>
        </>
      )}
    </div>
  );
}

export default Navbar;
