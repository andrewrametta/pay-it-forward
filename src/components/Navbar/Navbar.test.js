import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Navbar from "./Navbar";

test("renders Navbar", () => {
  render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>
  );
});
