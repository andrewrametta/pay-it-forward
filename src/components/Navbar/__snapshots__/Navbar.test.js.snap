// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders the complete page 1`] = `<Navbar />`;
